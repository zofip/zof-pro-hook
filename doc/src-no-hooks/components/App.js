import React, { Component } from 'react';

import jwt from 'jsonwebtoken';

import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

import AutocompleteTns from '../common/AutocompleteTns';
import ComponentGrid from './ComponentGrid';

import { APICLIENTS, APIPROJECTS, APICOMPONENTS } from '../utils/constants';

class App extends Component {

  state = {
    clients: [],
    projects: [],
    components: [],
    selectClient: null,
    selectProject: null
  };

  componentsTns= [];

  componentDidMount() {
    this.saveToken();
    this.getClients();
    this.project= null;
  }

  saveToken() {
    const token = jwt.sign({ admin: true, name: "admin" }, "T3ch&S0lv3-M0n1t0r", { expiresIn: 60 * 60 });
    localStorage.setItem("token", token);
  }

  getClients() {
    fetch(`${APICLIENTS}`, {
      method: "get",
      headers: {
        Authorization: "Bearer " + localStorage.getItem("token"),
        "Content-Type": "application/json"
      }
    })
      .then(result => result.json())
      .then(result => {
        this.setState({
          clients: result.map(res => {
            return { value: res.id, label: res.name };
          })
        })
      });
  }

  getProjectsByClient(client) {
    fetch(`${APIPROJECTS}?clientId=${client.value}`, {
      method: "get",
      headers: {
        Authorization: "Bearer " + localStorage.getItem("token"),
        "Content-Type": "application/json"
      }
    })
      .then(result => result.json())
      .then(result => {
        this.setState({
          projects: result.map(res => {
            return { value: res.id, label: res.name };
          })
        })
      });
  }

  getComponentsByProject(project){
    fetch(`${APICOMPONENTS}?projectId=${project.value}`, {
      method: "get",
      headers: {
        Authorization: "Bearer " + localStorage.getItem("token"),
        "Content-Type": "application/json"
      }
    })
      .then(result => result.json())
      .then(result => {
        this.componentsTns = result;
        this.setState({
          components: result
        })
      });
  }

  handleChangeClient = client => {
    this.setState({ selectClient: client });
    this.getProjectsByClient(client);
    this.project = null;
    this.componentsTns = [];
  };

  handleChangeProject = project => {
    this.setState({ selectProject: project });
    this.getComponentsByProject(project);
    this.project = project;
    this.componentsTns = [];
  };

  render() {
    const { clients, projects, selectClient } = this.state;
    
    return (
      <Container maxWidth="sm">
        <Box my={5}>
          <Typography variant="h4" component="h1" gutterBottom>
            Projects
          </Typography>
          <AutocompleteTns title="Clients" placeholder="Select Client ..."
            options={clients} handleChange={this.handleChangeClient} />
          <AutocompleteTns title="Projects" placeholder="Select Project ..." 
            options={projects} handleChange={this.handleChangeProject} 
            isDisabled={!selectClient} selectOption = {this.project}/>
          <ComponentGrid cardsComponent={this.componentsTns}></ComponentGrid>
        </Box>
      </Container>
    );
  }
}

export default App;