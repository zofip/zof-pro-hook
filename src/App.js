import React from 'react';

import { BrowserRouter as Router, Route } from "react-router-dom";

import Main from './pages/main/Main';
import IndicatorMain from "./pages/indicator/IndicatorMain";

const App = () => {

  return (
  <Router>
      <Route exact path="/" component={Main} />
      <Route path="/indicators/:componentId" component={IndicatorMain} />
  </Router>
  );
}

export default App;
