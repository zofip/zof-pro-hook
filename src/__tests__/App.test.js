import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('<App />', () => {
   it('Rendering', () => {
      const app = shallow(<App />);
      expect(app).toMatchSnapshot();
   });
});

// https://medium.com/@rossbulat/test-driven-development-in-react-with-jest-and-enzyme-2a6cf2cc3071
// https://itnext.io/testing-components-built-using-react-hooks-with-jest-enzyme-edb87d703756
// https://devhints.io/enzyme