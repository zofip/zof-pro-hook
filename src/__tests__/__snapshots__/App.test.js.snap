// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Rendering 1`] = `
<BrowserRouter>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/indicators/:componentId"
  />
</BrowserRouter>
`;
